src/Seabug/Contract/MarketPlaceSell.purs

```diff
@@ -0,0 +1,73 @@
+module Seabug.Contract.MarketPlaceSell
+  ( marketPlaceSell
+  ) where
+
+import Contract.Prelude
+
+import Contract.Address
+  ( getNetworkId
+  , ownPaymentPubKeyHash
+  , ownStakePubKeyHash
+  , payPubKeyHashBaseAddress
+  )
+import Contract.Monad (Contract, liftContractM, liftedE, liftedM)
+import Contract.PlutusData (toData)
+import Contract.ScriptLookups as Lookups
+import Contract.Transaction (TransactionHash, balanceAndSignTxE, submit)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value
+  ( CurrencySymbol
+  , TokenName
+  , Value
+  , singleton
+  , valueOf
+  )
+import Data.BigInt (fromInt)
+import Effect.Exception (throw)
+import Seabug.MarketPlace (marketplaceValidator)
+import Seabug.Types (MarketplaceDatum(..))
+
+-- | Mint the self-governed NFT for the given collection.
+marketPlaceSell
+  :: forall (r :: Row Type)
+   . CurrencySymbol /\ TokenName
+  -> Contract r TransactionHash
+marketPlaceSell (curr /\ tn) = do
+  owner <- liftedM "Cannot get PaymentPubKeyHash" ownPaymentPubKeyHash
+  ownerStake <- liftedM "Cannot get StakePubKeyHash" ownStakePubKeyHash
```

Comment:
Should it throw error when there is no staking key? Maybe constructing an address without staking key would be more desired

Reply:
Good point, copied this from the minting contract, ill update it